concepts/src/controllers/error.ts

```diff
@@ -0,0 +1,55 @@
+import { ErrorRequestHandler } from "express";
+
+export type ErrorResponse = {
+  httpStatus: number;
+  label: string;
+  description: string;
+  errorType: "http";
+  type: "Error";
+};
+
+export class HttpError extends Error {
+  public readonly status: number;
+  public readonly label: string;
+  public readonly description: string;
+
+  constructor({
+    status,
+    label,
+    description,
+  }: {
+    status: number;
+    label: string;
+    description: string;
+  }) {
+    super(description);
+    Object.setPrototypeOf(this, HttpError.prototype);
+
+    this.status = status;
+    this.label = label;
+    this.description = description;
+  }
+
+  get responseJson(): ErrorResponse {
+    return {
+      httpStatus: this.status,
+      label: this.label,
+      description: this.description,
+      errorType: "http",
+      type: "Error",
+    };
+  }
+}
+
+export const errorHandler: ErrorRequestHandler = (err, req, res, next) => {
+  const httpError =
+    err instanceof HttpError
+      ? err
+      : new HttpError({
+          status: 500,
+          label: "Server error",
+          description: "Expected server error: " + err.toString(),
+        });
```

Comment:
Minor: past convention has been to omit the `description` on 500 Internal Server Error, because from a user's POV we can't say anything useful there. I don't mind changing it, but want to check it's deliberate.

Reply:
This should also say "unexpected" 😬